src/sage/topology/simplicial_complex.py

```diff
@@ -3977,10 +4237,195 @@ def _enlarge_subcomplex(self, subcomplex, verbose=False):
+
+        ALGORITHM:
+
+        For each edge ``(x,y)`` in ``self``, find the links in ``self`` of
+        ``(x,y)``, ``(x)``, and ``(y)``.
+
+        Compute the intersection ``I`` of the links of ``(x)`` and ``(y)``.
+        The link ``L`` of ``(x,y)`` is a subcomplex of ``I``. Check whether
+        ``L`` is a deformation retract of ``I``. If so, the edge ``(x,y)``
+        can be contracted without altering the homotopy type of ``self``.
+
+        EXAMPLES::
+
+            sage: X = SimplicialComplex([[0,1], [1,2], [2,3], [3,4], [4,5], [5,0]])
+            sage: Y = simplicial_complexes.Sphere(1)
+            sage: X.pinch().is_isomorphic(Y)
+            True
+
+        ..SEEALSO::
+
+            :meth:`sage.topology.simplicial_complex.SimplicialComplex.link`,
+            :meth:`sage.topology.simplicial_complex.SimplicialComplex.reduce`
+        """
+        # The `lazy` shortcut never has downsides in dimension less than 4 so
+        # we even override the user in that case.
+        if self.dimension() < 4:
+            lazy = True
+        efset = frozenset()
+        X = self
+        renamed = set()
+        pinched = True
+        while pinched:
+            pinched = False
+            facets = {facet.set() for facet in X._facets}
+            for edge in X.faces()[1]:
+                edge = edge.set()
+                old, new = edge
+                if old in renamed or new in renamed:
+                    continue
+                old_set = {old}
+                new_set = {new}
+                edge_link_facets = []
+                old_link_facets = []
+                new_link_facets = []
+                for facet in facets:
+                    if edge <= facet:
+                        edge_link_facets.append(facet - edge)
+                    if old in facet:
+                        old_link_facets.append(facet - old_set)
+                    if new in facet:
+                        new_link_facets.append(facet - new_set)
+                edge_link = SimplicialComplex(edge_link_facets,
+                                              maximality_check=False)
+                intersection_faces = {of & nf
+                                      for of in old_link_facets
+                                      for nf in new_link_facets}
+                intersection = SimplicialComplex(intersection_faces)
+                if edge_link.is_deformation_retract(intersection,
+                                                    probable=lazy):
+                    pinched = True
+                    renamed.add(old)
+                    for facet in list(facets):
+                        if old in facet:
+                            facets.remove(facet)
+                            facet -= old_set
+                            facet |= new_set
+                            facets.add(facet)
+            if pinched:
+                X = SimplicialComplex(facets)
+        return X
+
+    def size(self):
```

Comment:
```suggestion
    def num_faces(self):
```
A more explicit name is better.